### PR TITLE
build & publish a pypi package

### PR DESCRIPTION
**Pull request by wiseaidev**
Completed tasks:

- [X] Added google style docs.
- [X] Refactored the `Lexer` class.
- [X] Added three functions to the lexer, namely `reset`, `skip_until`, and `skip_tokens`.
- [X] Migrated from `unittest` to `pytest`.
- [X] Added more unit tests, lint, and more.
- [X] Converted the Python project into a Pypi package hosted here:

[pypi version](https://pypi.org/project/deez_py/)

cc: @BatikanHyt

### CODE REVIEW

**Comment by BatikanHyt** [comment]
I really liked what you did. I changed project folder struce a bit and added pattern matching statement which introduced in python 3.10, please see #58 for the changes . There will be some conflicts after merge.

**Comment by wiseaidev** [comment]
Thanks for the review, @BatikanHyt! I am going to rebase once your PR is merged.

**Comment by vhladko** [comment]
@wiseaidev hey, hope you are doing great, any updates on this one ? :)

**Comment by wiseaidev** [comment]
Hey @vhladko, thanks for the update! I am going to sync these changes with the new branch. Gimme a moment!
P.S. Oh, wait... A parser has been implemented. Nice!

**Comment by vhladko** [comment]
@wiseaidev once you are done, pls ask @BatikanHyt to review it one more time, so we can move forward :)

**Comment by wiseaidev** [comment]
New changes introduced:
- [X]  Migrated `parser` tests from `unittest` to `pytest`.
- [X] Renamed the `ast` module to `deez_ast` because python3.10 has an `ast` causing a circular import.

cc: @BatikanHyt

**Comment by vhladko** [comment]
@BatikanHyt pls review when you have a minute :)

File: python/.pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
```
Comment (BatikanHyt) [inline python/.pre-commit-config.yaml:2]: I never use pre-commit-config but when lint it, it affects all the folders, not only python.

File: python/.pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    rev: v4.4.0
+    hooks:
+      - id: trailing-whitespace
+      - id: end-of-file-fixer
+      - id: mixed-line-ending
+        args: ["--fix=lf"]
+        description: Forces to replace line ending by the UNIX 'lf' character.
+  - repo: local
+    hooks:
```
Comment (BatikanHyt) [inline python/.pre-commit-config.yaml:11]: For all the hooks, it is unable to find the specified folders. I guess we need to specify the Python folder.

File: python/tox.ini [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+[testenv:lint]
+basepython=python3
+skip_install = true
+deps = pre-commit
+commands=
+    pre-commit run --all-files --show-diff-on-failure
+
+[testenv:test]
+whitelist_externals = poetry
```
Comment (BatikanHyt) [inline python/tox.ini:45]:
```suggestion
allowlist_externals = poetry
```
Maybe it's version-related, but I need to change to 'allowlist_external'.

File: python/tox.ini [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    PYTHONPATH = {toxinidir}
+deps = pytest-cov
+; If you want to make tox run the tests with the same versions, use poetry
+; to install the dependencies
+commands =
+    poetry install -v
+    poetry run pytest -v --cov=. tests/
+
+[testenv:coverage]
+whitelist_externals = poetry
```
Comment (BatikanHyt) [inline python/tox.ini:57]:
```suggestion
allowlist_externals = poetry
```

File: python/test/test_lexer.py [hunk 5]
```diff
@@ -1,123 +1,246 @@
-import unittest
```
Comment (BatikanHyt) [inline python/test/test_lexer.py:1]: The lint and test tool looks under the 'tests' directory, not 'test'.

File: python/Dockerfile [hunk 6]
```diff
@@ -3,4 +3,4 @@ FROM python:3.10-slim
 WORKDIR /app
 COPY . .
 
-ENTRYPOINT [ "python", "main.py" ]
\ No newline at end of file
+ENTRYPOINT [ "python", "deez_py" ]
```
Comment (BatikanHyt) [inline python/Dockerfile:6]:
```suggestion
ENTRYPOINT [ "python", "-m", "deez_py" ]
```
ModuleNotFoundError when running inside docker.

File: python/.pre-commit-config.yaml (hunk 1 above)
Comment (wiseaidev) [reply python/.pre-commit-config.yaml:2]: Oh wait, that's right, I forgot to run pre-commit. Imma fix it!

**Comment by wiseaidev** [comment]
Ready to merge!

**Comment by BatikanHyt** [comment]
Yes you can merge @vhladko

**Pull request merged**